mobile/lib/modules/home/ui/asset_grid/immich_asset_grid_view.dart

```diff
@@ -485,3 +323,292 @@ class ImmichAssetGridViewState extends State<ImmichAssetGridView> {
+              _Title(
+                selectionActive: selectionActive,
+                title: section.title!,
+                assets: scrolling
+                    ? []
+                    : renderList.loadAssets(section.offset, section.totalCount),
+                allAssetsSelected: allAssetsSelected,
+                selectAssets: selectAssets,
+                deselectAssets: deselectAssets,
+              ),
+            for (int i = 0; i < rows; i++)
+              scrolling
+                  ? _PlaceholderRow(
+                      key: ValueKey(i),
+                      number: i + 1 == rows
+                          ? section.count - i * assetsPerRow
+                          : assetsPerRow,
+                      width: width,
+                      height: width,
+                      margin: margin,
+                    )
+                  : _AssetRow(
+                      key: ValueKey(i),
+                      assets: assetsToRender.nestedSlice(
+                        i * assetsPerRow,
+                        min((i + 1) * assetsPerRow, section.count),
+                      ),
+                      absoluteOffset: section.offset + i * assetsPerRow,
+                      width: width,
+                      assetsPerRow: assetsPerRow,
+                      margin: margin,
+                      dynamicLayout: dynamicLayout,
+                      renderList: renderList,
+                      selectedAssets: selectedAssets,
+                      isSelectionActive: selectionActive,
+                      showStack: showStack,
+                      heroOffset: heroOffset,
+                      showStorageIndicator: showStorageIndicator,
+                      selectionActive: selectionActive,
+                      onSelect: (asset) => selectAssets([asset]),
+                      onDeselect: (asset) => deselectAssets([asset]),
+                    ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+/// The month title row for a section
+class _MonthTitle extends StatelessWidget {
+  final DateTime date;
+
+  const _MonthTitle({
+    required this.date,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final monthFormat = DateTime.now().year == date.year
+        ? DateFormat.MMMM()
+        : DateFormat.yMMMM();
+    final String title = monthFormat.format(date);
+    return Padding(
+      key: Key("month-$title"),
+      padding: const EdgeInsets.only(left: 12.0, top: 24.0),
+      child: Text(
+        title,
+        style: const TextStyle(
+          fontSize: 26,
+          fontWeight: FontWeight.w500,
+        ),
+      ),
+    );
+  }
+}
+
+/// A title row
+class _Title extends StatelessWidget {
+  final String title;
+  final List<Asset> assets;
+  final bool selectionActive;
+  final Function(List<Asset>) selectAssets;
+  final Function(List<Asset>) deselectAssets;
+  final Function(List<Asset>) allAssetsSelected;
```

Comment:
NIT: Can we make the return type more explicit for the `allAssetsSelected` callback?

```suggestion
  final bool Function(List<Asset>) allAssetsSelected;
```

Reply:
I've fixed this in my revision.